amqp-kit/client.go

```diff
@@ -311,3 +321,23 @@ func (c *Client) Close() error {
 
 	return nil
 }
+
+func (c *Client) checkExchange(channel *channel, exchange string) error {
+	c.exchLock.RLock()
+	_, exists := c.exchanges[exchange]
+	c.exchLock.RUnlock()
+	if exists {
+		return nil
+	}
+
+	err := channel.c.ExchangeDeclare(exchange, DefaultExchangeKind, true, false, false, false, nil)
```

Comment:
It is better to use `c.exchLock.Lock()` before ExchangeDeclare. Additionally you need to check exchanges map immediately after locking, to prevent concurrent calls of `channel.c.ExchangeDeclare` with the same parameters.